### PR TITLE
[api] Clean web3 code

### PR DESCRIPTION
**Pull request by Liuhaai**
`getStringAndBoolFromArray` and `getStringFromArray` are removed.

### CODE REVIEW

File: api/web3server_utils.go [hunk 1]
```diff
@@ -117,37 +117,6 @@ func intStrToHex(str string) (string, error) {
...
-	}
-	b, ok = params[1].(bool)
-	if !ok {
-		err = errUnkownType
-		return
-	}
-	return
-}
-
 func (svr *Web3Server) getBlockWithTransactions(blkMeta *iotextypes.BlockMeta, isDetailed bool) (blockObject, error) {
```
Comment (CoderZhi) [inline api/web3server_utils.go:120]: move this function back into web3server.go

**Review by CoderZhi — APPROVED** [review]

File: api/web3server_utils.go [hunk 2]
```diff
@@ -437,65 +406,48 @@ func parseLogRequest(in gjson.Result) (*filterObject, error) {
...
 
-func parseCallObject(in interface{}) (address.Address, string, uint64, *big.Int, []byte, error) {
+func parseCallObject(in *gjson.Result) (address.Address, string, uint64, *big.Int, []byte, error) {
 	var (
 		from     address.Address
-		to       string
-		gasLimit uint64
-		value    *big.Int
+		to       string   = ""
+		gasLimit uint64   = 0
```
Comment (dustinxie) [inline api/web3server_utils.go:413]: these 2 not necessary

File: api/web3server.go [hunk 3]
```diff
@@ -224,43 +223,47 @@ func (svr *Web3Server) handleWeb3Req(web3Req gjson.Result) interface{} {
 		res, err = svr.getHashrate()
 	case "eth_getLogs":
 		var filter *filterObject
-		if filter, err = parseLogRequest(web3Req.Get("params")); err == nil {
+		filter, err = parseLogRequest(web3Req.Get("params"))
+		if err == nil {
```
Comment (dustinxie) [inline api/web3server.go:227]: not necessary

File: api/web3server.go [hunk 4]
```diff
@@ -345,12 +348,12 @@ func (svr *Web3Server) getBlockNumber() (interface{}, error) {
 	return uint64ToHex(svr.coreService.TipHeight()), nil
 }
 
-func (svr *Web3Server) getBlockByNumber(in interface{}) (interface{}, error) {
-	blkNum, isDetailed, err := getStringAndBoolFromArray(in)
-	if err != nil {
-		return nil, err
+func (svr *Web3Server) getBlockByNumber(in *gjson.Result) (interface{}, error) {
+	blkNum, isDetailed := in.Get("params.0"), in.Get("params.1")
```
Comment (dustinxie) [inline api/web3server.go:352]: good clean-up

**Review by dustinxie — APPROVED** [review]

**Pull request merged**